### PR TITLE
Add atOrBefore and atOrAfter to DateTimeBasics

### PR DESCRIPTION
**Pull request by zdg2102**
Also update the internals of DateTimeBasics to equivalent implementations that let us make clearer guarantees about contract in the dartdoc.

### CODE REVIEW

File: lib/date_time_basics.dart [hunk 1]
```diff
@@ -7,40 +7,52 @@ extension DateTimeBasics on DateTime {
   /// Returns true if [this] occurs strictly before [other], accounting for time
   /// zones.
   ///
-  /// Delegates to [DateTime.compareTo] and therefore obeys the contract of that
-  /// method.
-  bool operator <(DateTime other) => compareTo(other) < 0;
+  /// Alias for [DateTime.isBefore].
+  bool operator <(DateTime other) => this.isBefore(other);
```
Comment (johnsonmh) [inline lib/date_time_basics.dart:11]: nit: I don't think you actually need the `this` on any of these.

**Review by johnsonmh — APPROVED** [review]
```
██╗      ██████╗ ████████╗███╗   ███╗
██║     ██╔════╝ ╚══██╔══╝████╗ ████║
██║     ██║  ███╗   ██║   ██╔████╔██║
██║     ██║   ██║   ██║   ██║╚██╔╝██║
███████╗╚██████╔╝   ██║   ██║ ╚═╝ ██║
╚══════╝ ╚═════╝    ╚═╝   ╚═╝     ╚═╝
```

File: lib/date_time_basics.dart (hunk 1 above)
Comment (zdg2102) [reply lib/date_time_basics.dart]: Yep, you're right. Removed now.

**Comment by jamesderlin** [comment]
Sorry that I didn't notice this change earlier.  What was wrong with the old `compareTo` implementation?  I'd expect that `isAtSameMomentAs(other) || isBefore(other);` probably involves more comparisons than `compareTo(other) <= 0`.  The [`DateTime.compareTo`](https://api.dart.dev/stable/2.13.2/dart-core/DateTime/compareTo.html) documentation already states:

> Returns a negative value if this DateTime isBefore `other`. It returns 0 if it isAtSameMomentAs `other`, and returns a positive value otherwise (when this isAfter `other`).

so I don't see a need for `DateTimeBasics` to implement the same guarantee independently.

Also, I see that #26 had the good intention of clarifying the docs, but it removed the statement:

 ```dart
/// Note that *unlike* the behavior of [DateTime.operator ==], the provided
/// comparison operators are time zone independent.
````

but I felt that it was important to indicate that the equality behaviors of `operator >=` and `operator <=` are intentionally *not* consistent with `operator ==`.

**Comment by zdg2102** [comment]
@jamesderlin basically the thought process was that `DateTime` already had these methods that provided the exact behavior we wanted the inequality operators to have, so it made more sense to treat the operators as aliases than to provide separate implementations with the goal of reproducing the same result.

As for #26, I see your point. I'll add something back to point out that mixing these operators with the equality operator is not going to do what it would naively look like it should do.

**Pull request merged**